examples/merge/parent.fmf

```diff
@@ -36,3 +36,11 @@ very:
                 x+: 1
                 y-: 1
                 z: 0
+
+# Regexp used to change values - note that the second tag regexp cannot be applied
+# as Tier2 is already modified into 't2'
+/regexp:
+    description~: "/.en(.*) .*/gen\\1/"
+    tags~:
+    - "/Tier(.)/t\\1/"
```

Comment:
Worth documenting the `\\1` vs `\1` pattern, and compare with `sed` and python `regex`. That would also help the reader get an intuition on how this is handled

Reply:
My intuition is that except for the main separator character everything would be parsable as native python-regex syntax, e.g. something like this should be parsable `'/(?i)prefix(?P<my_name>[\.\d]+)/replace_pattern/'` (i.e. to be able to toggle case-sensitivity, use named patterns etc.).

The replaced syntax might be tricky and I think `sed` does not allow named regex groups, but python supports it with `\g<my_name>`. Alternatively I was thinking `f-string` format would be intuitive, but maybe hard to implement.

Documentation wise I would say (just talking points, better paragraph should be written):
> The `~` find-and-replace pattern follows a `sed`-like pattern with python-native regex functions:
>
> `<separator><regex_search_pattern><separator><regex_replace_pattern><separator>`
>
> You may choose any `<separator>` character similar to `sed` format. Common patterns are `/`, `#`, `|`. Choose one that best fits your pattern. If necessary you can escape the separator by prepending `\` to it. Any regex pattern processing is only happening after the raw string is split by the separator.
> 
> The search and replace patterns are treated as python's `re.pattern` supporting any operations like `(?i)`, `(?P<named_group>)`, etc., and for the replace pattern you can use constructs like `\g<named_group>`, `\g<0>`

For yaml do we need to use `'` to better handle the string escapes?

Reply:
The `\\1` vs `\1` is just way how to get raw string into python, isn't it? So you are right - it needs to be `"\\1"` but can be `'\1'` (double vs single quotes).

Reply:
How about this docs:

The `~` text substitution operator evaluates `re.sub` function on each existing value of the key. The 'pattern' and 'repl' parameters have to be provided in the form of `<separator><pattern value><separator><repl value><separator>`. You may choose any character as the  separator, if such character is used in 'pattern' or 'repl' it will have to be escaped by prepending `\` to it. After the split of the substitution value into the pattern and repl arguments the separator character will be unescaped. Except of this no further processing of the substitution values happen.

See https://docs.python.org/3/library/re.html#re.sub for re.sub capabilities.  All occurrences of the pattern are replaced and pattern flags are not set.

For better readability use single quotes `'` in the fmf file, as they do not require the substitution value to be further escapes to meet the YAML syntax.

Reply:
> How about this docs:

Yep, the docs seem good. Just one note that `pattern` and `repl` are a bit specialized. It would be good to add something  like `'pattern' (as in search pattern)` and `'repl` (as in replace format)` to be more readable for newcomers.